src/ApplicationDefaultCredentials.php

```diff
@@ -136,7 +136,7 @@ public static function getMiddleware(
      * @param array $cacheConfig configuration for the cache when it's present
      * @param CacheItemPoolInterface $cache
      *
-     * @return CredentialsLoader
+     * @return CredentialsLoader|null
```

Comment:
So far as I can determine, this always returns a `CredentialsLoader` or throws an exception.

Reply:
You're right, good catch! I was blinded by `$creds` being initialized with  `null` and returned at the end, without checking in between 🙈. I reverted the change, thank you!